File: 334-IncreasingTripletSubsequence/334-IncreasingTripletSubsequence.py
# Last updated: 11/8/2025, 9:26:06 PM
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        smallest = float('inf')
        middle = float('inf')
      
        for current_num in nums:
            
            if current_num > middle:
                return True
          
           
            if current_num <= smallest:
                smallest = current_num
            else:
                
                middle = current_num
      
        
        return False

File: 334-IncreasingTripletSubsequence/334-IncreasingTripletSubsequence.cpp
// Last updated: 2/19/2026, 1:35:10 PM
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {

        if (nums.size() == 1 || nums.size() == 2){
            return false;
        }


        
        int first = INT_MAX;
        int second = INT_MAX;

        for (int i = 0; i < nums.size(); i++){
            if (nums[i] > second){
                return true;
            }
              
            else if (nums[i] <= first){
                first = nums[i];
            }
          else {
                second = nums[i];
            }
          
        }
    return false;

    }
};

